modules/distr-tracing-otel-config-instrumentation.adoc

```diff
@@ -0,0 +1,239 @@
+
+Auto-instrumentation in OpenTelemetry refers to the capability where the framework automatically instruments an application, without
+necessitating manual code changes. This enables developers and operators to get observability into their applications with minimal effort and changes to the existing codebase.
+
+== Important
+
+Note that the {OTELName} operator product only supports the injection mechanism of the instrumentation libraries: not the instrumentation libraries or the upstream images. Customers can build their own instrumentation images or use the ones provided by the community.
+
+== Instrumentation options
+
+_sample OpenTelemetry collector custom resource file_
+
+[source,java]
+----
+apiVersion: opentelemetry.io/v1alpha1
+kind: Instrumentation
+metadata:
+  name: java
+spec:
+  env:
+    - name: OTEL_TRACES_EXPORTER
+      value: otlp
+    - name: OTEL_EXPORTER_OTLP_TIMEOUT
+      value: "20"
+  exporter:
+    endpoint: http://production-collector.svc.cluster.local:4317
+  propagators:
+    - w3c
+  sampler:
+    type: parentbased_traceidratio
+    argument: "0.25"
+  java:
+    env:
+    - name: OTEL_JAVAAGENT_DEBUG
+      value: "true"
+----
+
+Table 1. Parameters used by the Operator to define the Instrumentation
+
+[cols=",,",options="header",]
+|===
+|Parameter |Description |Values
+|env |Common environment variables to define across all the instrumentations |
+|exporter |Exporter defines exporter configuration |
+|propagators |Propagators defines inter-process context propagation configuration |w3c;tracecontext;baggage;b3;none
+|resource |Resource attributes configuration |
+|sampler |Sampling configuration |
+|apacheHttpd |Configuration for the ApacheHTTPD instrumentation |
+|dotnet |Configuration for the DotNet instrumentation |
+|go |Configuration for the Golang instrumentation |
+|java |Configuration for the Java instrumentation |
+|nodejs |Configuration for the NodeJS instrumentation |
+|python |Configuration for the Python instrumentation |
+|===
+
+=== Configure the Apache HTTPD auto-instrumentation
+
+These are the parameters for the `+.spec.apacheHttpd+` field:
+
+[cols=",,",options="header",]
+|===
+|Name |Description |Default
+|attrs |Apache HTTPD agent specific attributes |
+|configPath |Location of Apache HTTPD server configuration|/usr/local/apache2/conf
+|env |Apache HTTPD specific env vars |
+|image |container image with Apache SDK and auto-instrumentation |
+|resourceRequirements |the compute resource requirements |
+|version |Apache HTTPD server version |2.4 |
+|===
+
+To enable the injection, you need to provide the following annotation to your PodSpec:
+
+[source,java]
+----
+instrumentation.opentelemetry.io/inject-apache-httpd: "true"
+----
+
+=== Configure the DotNet auto-instrumentation
```

Comment:
https://github.com/open-telemetry/opentelemetry-operator#opentelemetry-auto-instrumentation-injection 

see the comment for python